python/sglang/srt/layers/communicator.py

```diff
@@ -541,6 +541,10 @@ def _scatter_hidden_states(
             forward_batch.gathered_buffer[: forward_batch.input_ids.shape[0]],
             hidden_states,
         )
+        if hidden_states.data_ptr() == global_hidden_states.data_ptr():
```

Comment:
Can your if-statement match the assert present in `dp_scatter`:
```
assert (
            local_tokens.untyped_storage() is not global_tokens.untyped_storage()
        ), "aliasing between local_tokens and global_tokens not allowed"
```
So that we check the underlying storage.